src/components/SelectComponent.tsx

```diff
@@ -73,10 +76,22 @@ export class SelectComponent extends React.Component<InlineComponentProps, any>
             <DropdownButton bsSize="small" style={{ width: "auto" }} id={this.props.field.field + "_dropdown"}
                             title={foreignTitle}>
                 {
-                    _.map(optionsData, ((doc: any, index: any) =>
-                        <MenuItem onSelect={(eventKey: any) => this.select(this.props.field, eventKey)}
-                                  key={index}
-                                  eventKey={this.props.field.valueType === "object" ? doc : doc[this.props.field.foreign.key]}>{doc[this.props.field.foreign.title]}</MenuItem>))
+                    _.map(optionsData, ((doc: any, index: any) => {
+                        let eventKey = doc
+                        if (this.props.field.valueType === "object") {
```

Comment:
since eventKey is already doc ... is this if condition needed ?

Reply:
redundant, just keeping for clarity of logic(covering all cases)